### PR TITLE
Add Building an API pt. 10 Problem Set page

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
[Pt. 10 Asana ticket](https://app.asana.com/0/1205655776549324/1206697443678575/f)
[Pt. 10 Problem Set ticket](https://app.asana.com/0/1205655776549324/1207017852288093/f)

### CODE REVIEW

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- A list of any questions you weren't able to research independently
+
+## Practice
+
+Continue your independent Hello Books API.
+
+Fulfill the following requirements, with the same functionality presented in this topic:
+
+1. Recreate the following steps of the Hello Books repo:
+   1. Setup the `Genre` model with the attributes `id` and `name`. Implement `to_dict` and `from_dict` methods, as we did for the `Book` and `Author` models.
```
Comment (apradoada) [inline api-10-many-to-many/problem-set-many-to-many.md:21]:
```suggestion
   1. Set up the `Genre` model with the attributes `id` and `name`. Implement `to_dict` and `from_dict` methods, as we did for the `Book` and `Author` models.
```
Just a small grammatical change!

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   2. Create a migration to add a table for the `Genre` model and then apply it. Confirm that the `genre` table has been created as expected.
```
Comment (apradoada) [inline api-10-many-to-many/problem-set-many-to-many.md:22]:
```suggestion
   2. Create a migration to add a table for the `Genre` model and then apply it. Confirm that the `genre` table has been created in your database as expected.
```
I feel like the more explicit we are here, the less likely we are to have to field questions about where to find where the `genre` table should have been created. And then we don't have to repeat it later when they are asked to check to see if the `book_genre` table has been created.

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+   3. Create a join table to connect the `Book` and `Genre` models by creating a `BookGenre` model with 2 foreign key attributes `book_id` and `genre_id`.
+   4. Use SQLAlchemy relationship declarations to establish _optional_ attributes on the `Book` and `Genre` models to easily access the associated values from either side of the many-to-many relationship.
+   5. Create a migration to add the `book_genre` join table and then apply it. Confirm that the `book_genre` table has been created as expected.
+   6. Update the `Book` model's `from_dict` function to use value of the `genres` key from the `book_data` parameter if it exists, otherwise initialize the `genres` attribute as an empty list.
+   7. Update the `Book` model's `to_dict` function to include the names of any associated genres, only if such values exist.
+
+2. Create or refactor the following RESTful endpoints, with the same functionality presented in this topic:
```
Comment (apradoada) [inline api-10-many-to-many/problem-set-many-to-many.md:29]:
```suggestion
2. Create or refactor the following RESTful endpoints with the same functionality presented in the refactoring module:
```
I wasn't sure exactly what "topic" was referring to. I took a guess as to the specific module, but if I got it wrong, just updating it to something more specific could help here.

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+   1. As a client, I want to send a request with new valid genre data and get a success response containing the new genre's `id` and `name`, so that I know the API saved the genre data.
```
Comment (apradoada) [inline api-10-many-to-many/problem-set-many-to-many.md:30]:
```suggestion
   1. As a client, I want to send a request with new valid genre data. The success response should contain the new genre's `id` and `name` as a sign the API saved the genre data.
```
This is just a bit of a wording change to more clearly delineate what the request and response should contain.

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+   2. As a client, I want to send a request to get all existing genres, so that I can see a list of genres, with their `id` and `name`.
```
Comment (apradoada) [inline api-10-many-to-many/problem-set-many-to-many.md:31]:
```suggestion
   2. As a client, I want to send a request to get all existing genres. The response should list the genres with their `id` and `name`.
```

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+   3. As a client, I want to send a request with new valid book data and have it associated with a specific genre identified in the endpoint, then get a success response containing the new book's `id`, `title`, `description`, and the genre's name, so that I know the API saved the book data.
```
Comment (apradoada) [inline api-10-many-to-many/problem-set-many-to-many.md:32]:
```suggestion
   3. As a client, I want to send a request with new valid book data and have it associated with a specific genre identified in the endpoint. The success response should contain the new book's `id`, `title`, `description`, and the genre's `name` as a sign the API saved the book data.
```

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+   4. As a client, I want to send a request to get all books with an existing genre identified in the endpoint, so that I can see a list of books with their `id`, `title`, `description`, and if applicable, the author's name and all genres associated with the book.
```
Comment (apradoada) [inline api-10-many-to-many/problem-set-many-to-many.md:33]:
```suggestion
   4. As a client, I want to send a request to get all books with an existing genre identified in the endpoint. The response should contain a list of books with their `id`, `title`, `description`, and if applicable, the author's `name` and all genres associated with the book.
```

**Review by apradoada — COMMENTED** [review]
This looks good! I just added a couple of wording change suggestions to make things a little clearer. Feel free to use them or not! The biggest changes are in the endpoints that should be refactored. Reading through them, it was a little hard to parse through exactly what was expected, so I separated the request and response requirements into individual sentences.

File: api-10-many-to-many/problem-set-many-to-many.md (hunk 3 above)
Comment (kelsey-steven-ada) [reply api-10-many-to-many/problem-set-many-to-many.md:29]: Topic is meaning all the lessons for part 10 that are part of this tile...I'm wondering if "in this Learn tile" would be clearer? The endpoints are described in the last 2 of the many-to-many lessons in this tile.

File: api-10-many-to-many/problem-set-many-to-many.md (hunk 4 above)
Comment (kelsey-steven-ada) [reply api-10-many-to-many/problem-set-many-to-many.md:30]: Thank you, these definitely felt like run on sentences.

File: api-10-many-to-many/problem-set-many-to-many.md (hunk 3 above)
Comment (apradoada) [reply api-10-many-to-many/problem-set-many-to-many.md:29]: I think "in this Learn tile" works! I think that is closer to language we use elsewhere. We could also just go real simple with "in this Lesson"! I think either is a little clearer.

File: api-10-many-to-many/problem-set-many-to-many.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+   2. Create a migration to add a table for the `Genre` model and then apply it. Confirm that the `genre` table has been created in your database as expected.
+   3. Create a join table to connect the `Book` and `Genre` models by creating a `BookGenre` model with 2 foreign key attributes `book_id` and `genre_id`.
+   4. Use SQLAlchemy relationship declarations to establish _optional_ attributes on the `Book` and `Genre` models to easily access the associated values from either side of the many-to-many relationship.
+   5. Create a migration to add the `book_genre` join table and then apply it. Confirm that the `book_genre` table has been created as expected.
+   6. Update the `Book` model's `from_dict` function to use value of the `genres` key from the `book_data` parameter if it exists, otherwise initialize the `genres` attribute as an empty list.
+   7. Update the `Book` model's `to_dict` function to include the names of any associated genres, only if such values exist.
+
+2. Create or refactor the following RESTful endpoints, with the same functionality presented in this topic:
```
Comment (kelsey-steven-ada) [inline api-10-many-to-many/problem-set-many-to-many.md:29]:
```suggestion
2. Create or refactor the following RESTful endpoints, with the same functionality presented in this Learn tile:
```

**Review by apradoada — APPROVED** [review]
Looks good!
